pkg/exthttp/hedging.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package exthttp
+
+import (
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/caio/go-tdigest"
+	"github.com/cristalhq/hedgedhttp"
+)
+
+type CustomBucketConfig struct {
+	HedgingConfig HedgingConfig `yaml:"hedging_config"`
+}
+
+type HedgingConfig struct {
+	UpTo     uint    `yaml:"up_to"`
+	Quantile float64 `yaml:"quantile"`
+}
+
+var CustomBktConfig = CustomBucketConfig{
```

Comment:
It's not good to have "global" state. Let's move (copies of) this variable into where it is used. We could have a function like DefaultHedgingConfig() which would return a placeholder with defaults.